### ISSUE TITLE
syncthing-init.service: error while merging configuration due to api key reset

### ISSUE THREAD

**Issue by 3JlOy-PYCCKUi** [NixOS/nixpkgs#260262]
### Describe the bug
If you add enough (minimum number I haven't searched for) settings, then the script of applying these settings via api will not be able to execute completely and curl will generate "CSRF Error" error, this happens because syncthing resets api key after applying a certain number of settings

### Steps To Reproduce
1. Enable services.syncthing
2. Add several devices and options
3. Check logs of syncthing-init.service, there will be errors

### Expected behavior
syncthing-init.service will execute entirely with no errors and will apply the entire configuration without errors

### Screenshots
![errors in log](https://0x0.st/s/Mz_Wdez1efj5w3vyDkkDJQ/H4Zc.png)

### Additional context
I made a vm with the minimal configuration at which the error is reproduced
I used nixpkgs commit fdd898f8f79e8d2f99ed2ab6b3751811ef683242
Config for vm - https://0x0.st/s/ig6jujMXOGtBTFExhhW6nw/H4HC.nix
Steps to reproduce

1. Download nixpkgs - `curl -LO https://github.com/NixOS/nixpkgs/archive/fdd898f8f79e8d2f99ed2ab6b3751811ef683242.tar.gz`
2. Unpack - `tar -xf fdd898f8f79e8d2f99ed2ab6b3751811ef683242.tar.gz`
3. Download vm config - `curl -LO https://0x0.st/s/ig6jujMXOGtBTFExhhW6nw/H4HC.nix`
4. Build vm - `nixos-rebuild -I "nixpkgs=nixpkgs-fdd898f8f79e8d2f99ed2ab6b3751811ef683242" -I nixos-config=./H4HC.nix build-vm --no-flake`
5. Run vm, login as `test` with password `test`
6. Run `sudo systemctl status syncthing-init.service`
7. Get "CSRF Error" and parse error from jq

Also during testing I managed to fix this by parsing the api key before each curl call, this helps but looks like a dirty solution

### Notify maintainers

@doronbehar

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

**Comment by doronbehar** [comment]
Hmm indeed we recently changed a few details in the `syncthing-init.service` script in:

- https://github.com/NixOS/nixpkgs/pull/259259 

I tried to write a nixos test to test some of these issues, and I experienced numerous other bugs, and I sort of reproduced the issue as well. I Reverted

- #259259

In:

- https://github.com/NixOS/nixpkgs/pull/260303

And reopened:

- https://github.com/NixOS/nixpkgs/issues/259256

cc @Lassulus .

**Comment by doronbehar** [comment]
Could you try to merge https://github.com/NixOS/nixpkgs/pull/260303 locally and test?

**Comment by doronbehar** [comment]
OK. Thanks for the report!

**Comment by doronbehar** [comment]
> I also found that to add several different devices you need to have different id's, so I tested with four devices with different id's.

I also found out that during my testings in https://github.com/NixOS/nixpkgs/pull/260306 .

> With my patch, all 4 devices were added to config.xml correctly and syncthing-init.service ran correctly and without errors

You are invited to submit that patch, with better formatting. I am also looking into finding a way to add dummy, but valid devices ID, for testings.

**Comment by doronbehar** [comment]
OK I managed to fix the test, regarding checking that the script indeed deletes devices that need to be deleted. I haven't shared it in #260306, because I use there private (sort-of) syncthing's devices' IDs I took from my personal configuration. I'm still looking for a way to generate valid dummy devices IDs. My help request is at syncthing's forum:

https://forum.syncthing.net/t/how-to-generate-dummy-device-ids/20927

I also applied a better version of your patch to the (currently draft) #260306. I plan to add many more devices to that test, and maybe do the same for folders. Then I'll also test issue https://github.com/NixOS/nixpkgs/issues/259256 . In the meantime, I'm waiting for a dummy device ID generator.

**Comment by doronbehar** [comment]
@3JlOy-PYCCKUi I managed to create dummy syncthing device IDs and I am using them now in my PR #260306. I saw your PR https://github.com/NixOS/nixpkgs/pull/260556, and I think we can concentrate our efforts together on my PR at #260306, where I'm currently failing to reproduce your issue, your comments will be appreciated.

I am testing your issue without the 2nd commit in my PR (that is identical to your commit at #260556), and I'm testing that #259256 is fixed by reverting commit 82b59a8fb2331 just for testing (this works :)).
